daemon/snap.go

```diff
@@ -128,7 +128,7 @@ func allLocalSnapInfos(st *state.State, all bool, wanted map[string]bool) ([]abo
 		var err error
 		if all {
 			for _, seq := range snapst.Sequence {
-				info, err = snap.ReadInfo(seq.RealName, seq)
```

Comment:
Is real name the same as name in the snap state?

Reply:
Not always. `RealName` is from `snap.SideInfo` and comes directly from snap YAML file. In snapstate we're using `snap.Info.InstanceName()` (i.e. `name` at this location), which is `RealName` + optionally `InstanceKey`. That's what we should be using here.

Reply:
`RealName` as in the name blessed by the store, as opposed to the one in the snap YAML file that was merely requested.

Reply:
Missed how loading from the file only sets `SuggestedName`. Anyways, AFAICT `name` is the one we want to use here.